src/hotspot/share/code/icBuffer.hpp

```diff
@@ -62,16 +62,27 @@ class ICStub: public Stub {
   // General info
   int     size() const                           { return _size; }
 
-  // ICStub_from_destination_address looks up Stub* address from code entry address,
-  // which unfortunately means the stub head should be at the same alignment as the code.
-  static  int alignment()                        { return CodeEntryAlignment; }
+  // For extra correctness/safety, we want to make sure that each ICStub is in
+  // a separate instruction cache line. This would allow for piggybacking on instruction
+  // cache coherency on some architectures to order the updates to ICStub and setting
+  // the destination to the ICStub. Note that cache lines size might be larger than
+  // CodeEntryAlignment that is a normal alignment for CodeBlobs.
```

Comment:
This is a paragraph that raises more questions than it answers. The relationship between ordering and instruction cache line coherency is pretty tenuous even on x86 these days, isn't it?  I'd not say "For extra correctness/safety," but "To be cautious,".

Reply:
Sure. How does the new comment look?